src/app/labeling/qgsmaptoollabel.cpp

```diff
@@ -939,7 +1087,7 @@ bool QgsMapToolLabel::createAuxiliaryFields( LabelDetails &details, QgsPalIndexe
     {
       index = vlayer->fields().lookupField( prop.field() );
     }
-    else
+    else if ( prop.propertyType() != QgsProperty::ExpressionBasedProperty )
```

Comment:
This is handled in QgsAuxiliaryLayer::createProperty -- there's logic in there which upgrades an expression based property to allow for combined handling of expressions AND manually placed labels.